### PR TITLE
[e2e] e2e web changes part1: registering web extension.

### PR DESCRIPTION
**Pull request by nturgut**
## Description

In order to use different e2e package for Flutter Web engine integration tests we need to add some
web specific parts to the code. These can be found here as a demo: [PR](https://github.com/flutter/flutter/pull/49833)

In this particular change we are registering flutter driver commands to the browser. They will be accessed as window.$flutter_command

## Related Issues

https://github.com/flutter/flutter/issues/50212

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/e2e/lib/_extension_io.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// The dart:io implementation of [registerWebServiceExtension].
```
Comment (yjbanov) [inline packages/e2e/lib/_extension_io.dart]: nit: enclose references to symbols in backticks, e.g. `dart:io`

File: packages/e2e/lib/_extension_web.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/// The dart:html implementation of [registerWebServiceExtension].
+///
+/// Registers Web Service Extension for Flutter Web application.
+///
+/// window.$flutterDriver will be called by Flutter Web Driver to process
+/// Flutter command.
+///
+/// See also:
+///
+///  * [_extension_io.dart], which has the dart:io implementation
```
Comment (yjbanov) [inline packages/e2e/lib/_extension_web.dart]: ditto: `dart:io`

**Comment by collinjackson** [comment]
> Can you write a test for this change?

Note that the E2E plugin is currently tested using a cross-platform Dart integration test that lives here: https://github.com/flutter/plugins/blob/master/packages/e2e/example/test_driver/example_e2e.dart

So, enabling the integration test for the browser platform on the flutter/plugins CI would result in functional testing of web support. This is something that would happen by changing the plugin repo's Cirrus configuration: https://github.com/flutter/plugins/blob/master/.cirrus.yml

That said, unit tests would obviously be useful as well.

File: packages/e2e/lib/_extension_io.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+///
+/// See also:
+///
+///  * [_extension_web.dart], which has the dart:html implementation
+void registerWebServiceExtension(
+    Future<Map<String, dynamic>> Function(Map<String, String>) call) {
+  throw UnsupportedError('Use registerServiceExtension instead');
```
Comment (yjbanov) [inline packages/e2e/lib/_extension_io.dart]: Is this error message correct? We either call both `registerWebServiceExtension` _and_ `registerServiceExtension`, or we only call `registerServiceExtension`.

If I'm reading the code correct the error should read something like "registerWebServiceExtension is only available on the Web".

File: packages/e2e/lib/_extension_web.dart [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:html' as html;
+import 'dart:js';
+import 'dart:js_util' as js_util;
+
+/// The dart:html implementation of [registerWebServiceExtension].
```
[the remark this answers is not in the document]
Comment (nturgut) [reply packages/e2e/lib/_extension_web.dart]: done

File: packages/e2e/lib/_extension_web.dart [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+///
+/// Registers Web Service Extension for Flutter Web application.
+///
+/// window.$flutterDriver will be called by Flutter Web Driver to process
```
[the remark this answers is not in the document]
Comment (nturgut) [reply packages/e2e/lib/_extension_web.dart]: done

File: packages/e2e/lib/_extension_web.dart (hunk 2 above)
Comment (nturgut) [reply packages/e2e/lib/_extension_web.dart]: done.

File: packages/e2e/lib/_extension_io.dart (hunk 1 above)
Comment (nturgut) [reply packages/e2e/lib/_extension_io.dart]: done.

File: packages/e2e/lib/_extension_io.dart (hunk 3 above)
Comment (nturgut) [reply packages/e2e/lib/_extension_io.dart]: done.

**Comment by nturgut** [comment]
> Can you write a test for this change?

I think we can't write a unit test unless we use a tool like felt. Since we need the browser to tests the main functionality. We need what felt does such as downloading chrome and testing the file with it.

**Comment by yjbanov** [comment]
I think it's fine to use a system-installed Chrome for testing the test harness itself.

**Pull request merged**